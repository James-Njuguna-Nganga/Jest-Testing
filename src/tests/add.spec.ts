import { multiply } from '../services/add.service';

test('multiplies numbers', () => {
    expect.assertions(4)
  expect(multiply(2, 3)).toEqual(6);
  expect(multiply(3, 3)).toEqual(9);
  expect(multiply(4, 3)).toEqual(12);
  expect(multiply(5, 3)).toEqual(15);
});

test('complex object snapshot', () => {
  const user = {
    name: 'jimmy',
    age: 20,
    address:{ street:'12222 nyeri'}
  };
  expect(user).toMatchSnapshot({
    age: expect.any(Number),
    address: {
      street: expect.any(String)
    }
  })
})