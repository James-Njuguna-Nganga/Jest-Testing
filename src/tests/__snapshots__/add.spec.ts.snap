// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`complex object snapshot 1`] = `
{
  "address": {
    "street": Any<String>,
  },
  "age": Any<Number>,
  "name": "jimmy",
}
`;
